consensus/src/backup/saver.rs

```diff
@@ -10,30 +14,29 @@ const LOG_TARGET: &str = "AlephBFT-backup-saver";
 /// Component responsible for saving units into backup.
 /// It waits for items to appear on its receivers, and writes them to backup.
 /// It announces a successful write through an appropriate response sender.
-pub struct BackupSaver<H: Hasher, D: Data, S: Signature, W: AsyncWrite> {
-    units_from_runway: Receiver<UncheckedSignedUnit<H, D, S>>,
-    responses_for_runway: Sender<UncheckedSignedUnit<H, D, S>>,
+pub struct BackupSaver<H: Hasher, D: Data, MK: MultiKeychain, W: AsyncWrite> {
+    units_from_runway: Receiver<DagUnit<H, D, MK>>,
+    responses_for_runway: Sender<DagUnit<H, D, MK>>,
     backup: Pin<Box<W>>,
 }
 
-impl<H: Hasher, D: Data, S: Signature, W: AsyncWrite> BackupSaver<H, D, S, W> {
+impl<H: Hasher, D: Data, MK: MultiKeychain, W: AsyncWrite> BackupSaver<H, D, MK, W> {
     pub fn new(
-        units_from_runway: Receiver<UncheckedSignedUnit<H, D, S>>,
-        responses_for_runway: Sender<UncheckedSignedUnit<H, D, S>>,
+        units_from_runway: Receiver<DagUnit<H, D, MK>>,
+        responses_for_runway: Sender<DagUnit<H, D, MK>>,
         backup: W,
-    ) -> BackupSaver<H, D, S, W> {
+    ) -> BackupSaver<H, D, MK, W> {
         BackupSaver {
             units_from_runway,
             responses_for_runway,
             backup: Box::pin(backup),
         }
     }
 
-    pub async fn save_item(
-        &mut self,
-        item: &UncheckedSignedUnit<H, D, S>,
-    ) -> Result<(), std::io::Error> {
-        self.backup.write_all(&item.encode()).await?;
+    pub async fn save_unit(&mut self, unit: &DagUnit<H, D, MK>) -> Result<(), std::io::Error> {
+        let parents = unit.parents().clone();
+        let unit: UncheckedSignedUnit<_, _, _> = unit.clone().unpack().into();
```

Comment:
I think there's too much hustle here to "deconstruct" the `DagUnit`. With all of this cloning, we have 3 copies of `parents` in this scope. Maybe `DagUnit` needs a `fn unit()` which only clones the inner unit?

Reply:
Ah, I didn't notice this comment still makes sense after the above change (even though now it's only one unnecessary `clone` of parents). Hm, I would prefer to keep the interface simpler, but I guess I can add a `unit` method to `WrappedUnit`, although I'm not convinced the performance worries are that important here – you still need a `clone` of a similarly structured object, no?

Reply:
Ah, I didn't notice this comment still makes sense after the above change (even though now it's only one unnecessary `clone` of parents). Hm, I would prefer to keep the interface simpler, but I guess I can add a `unit` method to `WrappedUnit`, although I'm not convinced the performance worries are that important here – you still need a `clone` of a similarly structured object, no?

Reply:
how about adding it directly to `ReconstructedUnit` without changing `WrappedUnit`?

Reply:
I don't like the unnecessary clone, but if it's more convenient for your abstractions, then it's fine. I'll let you decide and resolve this comment.